File: android/ExampleApp/app/src/androidTest/java/com/fjuul/sdk/android/ExampleApp/ExampleUiTest.java
package com.fjuul.sdk.android.ExampleApp;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.filters.LargeTest;
import androidx.test.rule.ActivityTestRule;

@RunWith(AndroidJUnit4.class)
@LargeTest
public class ExampleUiTest {

    @Rule
    public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class);

    @Test
    public void assertSDKText_mainActivity() {
        onView(withId(R.id.main_text)).check(matches(withText("FJUUL_SDK")));
    }
}

File: android/ExampleApp/app/src/androidTest/java/com/fjuul/sdk/android/exampleapp/ExampleUiTest.kt
package com.fjuul.sdk.android.exampleapp

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.hasDescendant
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import androidx.test.rule.ActivityTestRule
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import com.fjuul.sdk.android.exampleapp.R as appR

@RunWith(AndroidJUnit4::class)
@LargeTest
class ExampleUiTest {
    @get:Rule
    var activityRule = ActivityTestRule(
        MainActivity::class.java
    )

    @Test
    fun shouldStartFromOnboarding() {
        onView((withId(appR.id.toolbar))).check(matches(hasDescendant(withText("Onboarding"))))
    }
}
